### PR TITLE
feat(*)!: Updates pkg-loader to use oci-wasm crate

### PR DESCRIPTION
**Pull request by thomastaylor312**
This adds support for the new oci-wasm crate and uses the types and client wrapper from that crate. This is marked as a breaking change because it will only support the new layer content type of `application/wasm` rather than the older format. I figured that would be fine since this is a new crate.

### CODE REVIEW

**Comment by thomastaylor312** [comment]
If we'd like to make it backwards compatible to the old layer and config type, we could probably add that in the crate as an option, but I figure it might be better to rip the bandaid

**Comment by thomastaylor312** [comment]
I'll figure out the test thing soon!

File: crates/wasm-pkg-loader/.vscode/settings.json
Comment (thomastaylor312) [inline crates/wasm-pkg-loader/.vscode/settings.json:1]: Note that I removed this directory because it was IDE specific and added it to gitignore

**Review by lann — APPROVED** [review]
lgtm. I'd just like to republish the bytecodealliance.org packages with the correct config before merging.

**Comment by lann** [comment]
> If we'd like to make it backwards compatible to the old layer and config type, we could probably add that in the crate as an option, but I figure it might be better to rip the bandaid

Nah. That was all just hedging my bets before the cncf thing had solidified.

**Review by calvinrp — APPROVED** [review]

**Comment by thomastaylor312** [comment]
Sounds great @lann. Is there a problem with merging and then not cutting a release until you republish those packages?

**Comment by lann** [comment]
Just that I might forget :sweat_smile:

**Comment by lann** [comment]
I'll either fix the packages or set reminder(s) for myself and merge anyway by eod tomorrow.

**Comment by lann** [comment]
Oh it looks like the oci-wasm crate/repo is still just owned by you @thomastaylor312; I think we should get those under some shared ownership before adding it as a dependency here to make sure we can land bug fixes etc.

**Comment by lann** [comment]
Also @thomastaylor312 I'm looking at updating those packages; how have you been testing this? Just `oras`?

**Comment by thomastaylor312** [comment]
@lann I already added @jsturtevant as another maintainer on the project and [I finished all the steps](https://github.com/bytecodealliance/governance/issues/88#issuecomment-2142620863) after approval to move to BCA. So I think we're ok there at this point

As for pushing, I've been using the latest main of `wash` to push things since it can push any component. You can also use the code I did in the e2e tests in this PR if you want to whip up something quick. Otherwise, I am working on adding push in #31

**Comment by lann** [comment]
> @lann I already added @jsturtevant as another maintainer on the project

Cool; could you add him as a crate owner too?

**Comment by thomastaylor312** [comment]
Yeah I can. I was mostly waiting so I could make a GH team have permissions, but I can do that for now. I sent him the invite and he should be accepting soon

**Comment by jsturtevant** [comment]
I've been added as a crate owner.  Thanks!

**Pull request merged**
